aten/src/ATen/SparseTensorUtils.h

```diff
@@ -11,16 +12,17 @@ using LongTensor = Tensor;
 using IntTensor = Tensor;
 using SparseType = Type;
 
-// This is an internal utility function for getting at the SparseTensorImpl,
+// This is an internal utility function for getting at the SparseCOOTensorImpl,
 // so that we can write sparse tensor specific accessors for special fields
 // in SparseTensor.  You should only use this for writing low level
-// setters/getters for SparseTensorImpl fields; otherwise, you should use
+// setters/getters for SparseCOOTensorImpl fields; otherwise, you should use
 // the low level setters/getters that were implemented using this.
 //
 // This may be called repeatedly, so make sure it's pretty cheap.
-inline SparseTensorImpl* get_sparse_impl(const SparseTensor& self) {
-  AT_ASSERTM(self.is_sparse(), "_internal_get_SparseTensorImpl: not a sparse tensor");
-  return static_cast<SparseTensorImpl*>(self.unsafeGetTensorImpl());
+template <typename SparseType=SparseCOOTensorImpl>
+inline SparseType* get_sparse_impl(const SparseTensor& self) {
+  AT_ASSERTM(self.is_sparse(), "_internal_get_SparseCOOTensorImpl: not a sparse tensor");
+  return static_cast<SparseType*>(self.unsafeGetTensorImpl());
 }
```

Comment:
Yes, so as I mentioned in the high level, there's no point in trying to template this. In fact, you have introduced a bug in the process of templating this: if I `get_sparse_impl<SparseCOOTensorImpl>`, my assert test should be asserting that I actually have a *COO tensor*. I'm not sure what `is_sparse()` means now, but I know there is no way for it to mean something and for `SparseGCSTensorImpl` to also work.

Reply:
Makes sense. Will implement a new function.